### PR TITLE
Add Team Cape 0/X/I to clue step

### LINKED ISSUE #17269: Team cape x, i, and zero not recognized for emote clue step
### Type

Incorrect behavior

### Bug description

When completing the clue for the following text, the overlay does not identify the Team cape x, Team cape i, or Team cape zero as valid for it, though they are usable for the step and can be deposited into the stash unit.

> Cheer in the Barbarian Agility Arena. Headbang before you talk to me.
>
> Equip a steel platebody, maple shortbow and a Wilderness cape.

### PR DESCRIPTION
**Pull request by Yenof**
Hopefully correctly added TEAM_CAPE_ZERO, TEAM_CAPE_X, and TEAM_CAPE_I to the Barbarian Agility Arena and Castle Wars clues. 

First RL PR, apologies for anything done incorrectly.

Fixes #17269

### CODE REVIEW

**Review by Felanbird — APPROVED** [review]
works as expected, tested with `TEAM_CAPE_I` & `TEAM46_CAPE`

**Comment by Nightfirecat** [comment]
This change looks good now. Can you also apply it to [another clue step I found here](https://github.com/runelite/runelite/issues/17269#issuecomment-1863878216)? At this point, it may be worth [defining an `AnyRequirementCollection` var for "Any team cape"](https://github.com/Yenof/runelite/blob/4a720dd805ca8952b77cda9d3364bf2061091c34/runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java#L86) so this particular code doesn't have to be repeated in multiple places.

**Comment by Yenof** [comment]
I went ahead and added it again in the current format to the other clue step. 
I looked into the AnyRequirementsCollection briefly and couldn't get it working very quickly. 
I want to look into it more, but Leagues atm. So I figured I'd just do the current solution for now. :D

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java [hunk 1]
```diff
@@ -95,6 +95,14 @@ public class EmoteClue extends ClueScroll implements LocationClueScroll
 		ItemVariationMapping.getVariations(PHARAOHS_SCEPTRE).stream()
 			.map(ItemRequirements::item)
 			.toArray(SingleItemRequirement[]::new));
+	private static final AnyRequirementCollection ANY_TEAM_CAPE = any("Any team cape",
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java:98]: isn't this just

```java
			Stream.of(TEAM_CAPE_I, TEAM_CAPE_X, TEAM_CAPE_ZERO, ItemVariationMapping.getVariations(TEAM1_CAPE).stream())
			.reduce(Stream::concat)
			.orElseGet(Stream::empty)
			.map(ItemRequirements::item)
			.toArray(SingleItemRequirement[]::new));
```
?

**Pull request merged**